Automattic/wp-calypso
# Signup: Use shopping cart manager

#### Changes proposed in this Pull Request

One of two remaining places in calypso which does not use the `@automattic/shopping-cart` package to interact with the `/me/shopping-cart` endpoint is signup. This PR modifies signup so that it uses the shopping-cart package instead. This is an alternative to https://github.com/Automattic/wp-calypso/pull/54262

Besides providing consistency with other cart management in calypso and separating the cart logic away from the signup infrastructure, this also allows us to remove `lib/signup/cart` which is a very old and poorly maintained cart management API that has caused us issues in the past.

Depends on https://github.com/Automattic/wp-calypso/pull/54667 which adds the `cartManagerClient` singleton which can be used to manipulate the cart outside of React.

#### Testing instructions

- Visit `/start` and add a paid domain and a paid plan.
- Verify there are no errors.
- Verify that when you reach checkout, both the domain and the plan are in your cart.
- Repeat the above steps with `/start/domain` and selecting just a paid domain.
